### PR TITLE
Release 4.8.0

### PR DESCRIPTION
**Pull request by MikaelMayer**
### Description
Merging the release 4.8.0 branch into master It brings several modifications that were deemed necessary
- Updated the doo files from the DafnyStandardLibraries,
- Updated doo files and dtr files from the test suite
- Renamed dfy files for externs so that they are in the same order and don't cause issue as a workaround for https://github.com/dafny-lang/dafny/issues/5728 and updated README about the naming of those files
- Added %rm %mv and %cp command which are used in tests but not available on every windows machine (mine included)
- Added support for any exit code so that `%rm` won't fail if a file or a folder does not exist (which is the case the first time)
- Added ability to run a command within a given folder as some commands would not run on my Windows.
- Added `-f` option to cp and mv to be able to run the same test multiple times
- Added missing sources to recompile multiple doo files in the test suite.

### How has this been tested?
Tests are passing on the CI and previously offending tests are also passing on my Windows machine.

<small>By submitting this pull request, I confirm that my contribution is made under the terms of the [MIT license](https://github.com/dafny-lang/dafny/blob/master/LICENSE.txt).</small>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -93,11 +93,15 @@ update-go-module:
...
+	(cd "${DIR}"; cd Source/DafnyStandardLibraries; make update-binary)
+
 # `make pr` will bring you in a state suitable for submitting a PR
 # - Builds the Dafny executable
 # - Use the build to convert core .dfy files to .cs
 # - Rebuilds the Dafny executable with this .cs files
 # - Apply dafny format on all dfy files
 # - Apply dotnet format on all cs files except the generated ones
 # - Rebuild the Go and C# runtime modules as needed.
+# - Rebuild the standard libraries
```
Comment (MikaelMayer) [inline Makefile:106]:
```suggestion
```

File: Makefile [hunk 2]
```diff
@@ -93,6 +93,9 @@ update-go-module:
 update-runtime-dafny:
 	(cd "${DIR}"; cd Source/DafnyRuntime/DafnyRuntimeDafny; make update-go)
 
+update-standard-libraries:
```
Comment (atomb) [inline Makefile:96]: Good thing to add!

File: Source/XUnitExtensions/Lit/ExitCommand.cs [hunk 3]
```diff
@@ -1,22 +1,32 @@
...
   }
 
   public async Task<int> Execute(TextReader inputReader,
     TextWriter outputWriter, TextWriter errorWriter) {
-    var exitCode = await operand.Execute(inputReader, outputWriter, errorWriter);
-    if (exitCode == expectedExitCode) {
+    var exitCode = 1;
+    try {
+      exitCode = await operand.Execute(inputReader, outputWriter, errorWriter);
+    } catch (Win32Exception) {
```
Comment (atomb) [inline Source/XUnitExtensions/Lit/ExitCommand.cs:22]: I suspect this exception would be a different one on non-Windows platforms. Or is it used on all platforms in .NET?

**Review by atomb — CHANGES_REQUESTED** [review]
There's one exception that I'm not sure is cross-platform (though maybe it is?) and a couple of test failures. Otherwise, it looks good!

File: Source/XUnitExtensions/Lit/ExitCommand.cs (hunk 3 above)
Comment (MikaelMayer) [reply Source/XUnitExtensions/Lit/ExitCommand.cs:22]: Windows does not have the "rm" command so I'm not sure if the original returns an exception if the file does not exist or not.

Comment (MikaelMayer) [reply Source/XUnitExtensions/Lit/ExitCommand.cs:22]: Note: the code was working fine in other platforms without the try catch, so I'm not expecting this to break any other platform.

**Review by atomb — DISMISSED** [review]

**Comment by ssomayyajula** [comment]
Amusingly, sorting gives a deterministic (and correct) order *except for* `py`, because `notarget < py`. Going to do some renaming.

```
notarget:
./Concurrent-notarget-cs.dfy
./Concurrent-notarget-java-cs-js-go-py.dfy
./FileIO3InternalExterns-notarget-java-cs-js-go-py.dfy
./FileIO5-notarget-java-cs-js-go-py.dfy

cs:
./Concurrent-notarget-cs.dfy
./Concurrent-notarget-java-cs-js-go-py.dfy
./FileIO2InternalExterns-java-cs-js.dfy
./FileIO3InternalExterns-notarget-java-cs-js-go-py.dfy
./FileIO5-notarget-java-cs-js-go-py.dfy

java:
./Concurrent-java.dfy
./Concurrent-notarget-java-cs-js-go-py.dfy
./FileIO2InternalExterns-java-cs-js.dfy
./FileIO3InternalExterns-notarget-java-cs-js-go-py.dfy
./FileIO5-notarget-java-cs-js-go-py.dfy

js:
./Concurrent-js.dfy
./Concurrent-notarget-java-cs-js-go-py.dfy
./FileIO2InternalExterns-java-cs-js.dfy
./FileIO3InternalExterns-notarget-java-cs-js-go-py.dfy
./FileIO5-notarget-java-cs-js-go-py.dfy

go:
./Concurrent-go.dfy
./Concurrent-notarget-java-cs-js-go-py.dfy
./FileIO1InternalExterns-go.dfy
./FileIO3InternalExterns-notarget-java-cs-js-go-py.dfy
./FileIO5-notarget-java-cs-js-go-py.dfy

py:
./Concurrent-notarget-java-cs-js-go-py.dfy
./Concurrent-py.dfy
./FileIO3InternalExterns-notarget-java-cs-js-go-py.dfy
./FileIO4InternalExterns-py.dfy
./FileIO5-notarget-java-cs-js-go-py.dfy
```

**Comment by MikaelMayer** [comment]
Changes accepted.

**Review by ssomayyajula — APPROVED** [review]

**Pull request merged**